modules/core/src/smithy4s/Service.scala

```diff
@@ -34,33 +34,48 @@ import kinds._
   *   around makes it drastically easier to implement logic generically, without involving
   *   metaprogramming.
   */
-trait Service[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]] extends FunctorK5[Alg] with Service.Provider[Alg, Op] {
-  implicit val serviceInstance: Service[Alg, Op] = this
-  val service = this
+trait Service[Alg[_[_, _, _, _, _]]] extends FunctorK5[Alg] with Service.Provider[Alg] {
+  type Operation[I, E, O, SI, SO]
+  type Endpoint[I, E, O, SI, SO] = smithy4s.Endpoint[Operation, I, E, O, SI, SO]
+  type Interpreter[F[_, _, _, _, _]] = PolyFunction5[Operation, F]
+  type FunctorInterpreter[F[_]] = PolyFunction5[Operation, kinds.Kind1[F]#toKind5]
+  type BiFunctorInterpreter[F[_, _]] = PolyFunction5[Operation, kinds.Kind2[F]#toKind5]
 
-  def endpoints: List[Endpoint[Op, _, _, _, _, _]]
-  def endpoint[I, E, O, SI, SO](op: Op[I, E, O, SI, SO]): (I, Endpoint[Op, I, E, O, SI, SO])
+  val service: Service[Alg] = this
+  def endpoints: List[Endpoint[_, _, _, _, _]]
+  def endpoint[I, E, O, SI, SO](op: Operation[I, E, O, SI, SO]): (I, Endpoint[I, E, O, SI, SO])
   def version: String
   def hints: Hints
-  def reified: Alg[Op]
-  def fromPolyFunction[P[_, _, _, _, _]](function: PolyFunction5[Op, P]): Alg[P]
-  def toPolyFunction[P[_, _, _, _, _]](algebra: Alg[P]): PolyFunction5[Op, P]
+  def reified: Alg[Operation]
+  def fromPolyFunction[P[_, _, _, _, _]](function: PolyFunction5[Operation, P]): Alg[P]
+  def toPolyFunction[P[_, _, _, _, _]](algebra: Alg[P]): PolyFunction5[Operation, P]
 
-  final val opToEndpoint : PolyFunction5[Op, Endpoint[Op, *, *, *, *, *]] = new PolyFunction5[Op, Endpoint[Op, *, *, *, *, *]]{
-    def apply[I, E, O, SI, SO](op: Op[I,E,O,SI,SO]): Endpoint[Op,I,E,O,SI,SO] = endpoint(op)._2
+  final val opToEndpoint : PolyFunction5[Operation, Endpoint] = new PolyFunction5[Operation, Endpoint]{
+    def apply[I, E, O, SI, SO](op: Operation[I,E,O,SI,SO]): Endpoint[I,E,O,SI,SO] = endpoint(op)._2
   }
 
 }
 
 object Service {
-  trait Provider[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]] extends HasId {
-    def service: Service[Alg, Op]
+
+  def apply[Alg[_[_, _, _, _, _]]](implicit ev: Service[Alg]): ev.type = ev
+
+  type Aux[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]] = Service[Alg]{ type Operation[I, E, O, SI, SO] = Op[I, E, O, SI, SO] }
```

Comment:
the `Aux` pattern allows to "recover" the type parameter that was moved, in locations where it matters